comp/netflow/server/server.go

```diff
@@ -30,29 +32,56 @@ type dependencies struct {
 	Hostname  hostname.Component
 }
 
+// NetflowServerStatus represents the status of the server including details about
+// listeners which are working and those which have closed.
+type NetflowServerStatus struct {
+	TotalListeners         int
+	OpenListeners          int
+	ClosedListeners        int
+	WorkingListenerDetails []NetflowListenerStatus
+	ClosedListenerDetails  []NetflowListenerStatus
+}
+
+// NetflowListenerStatus handles logic related to pulling config information and associating it to an error.
+type NetflowListenerStatus struct {
+	Config nfconfig.ListenerConfig
+	Error  string
+}
+
+// globalServer is only used on getting the status of the server.
+var (
+	globalServer   = &Server{}
+	globalServerMu sync.Mutex
+)
```

Comment:
I think it might be safer to use `expvar` to store the status information instead of relying on a globalServer structure.

1/ We are using mutex around assignment of server to globalServer, but it's only protecting at startup time.
But after the startup, the server object is access in `GetStatus`, but can also be modified elsewhere in NetFlow server runtime.

2/ I think it might be safer to store NetFlow status information in a expvar like this example from compliance product: https://github.com/DataDog/datadog-agent/blob/5468729c4c5429df38b81e9b4260bbaab958cc20/pkg/compliance/agent.go#L34

3/ Global should be avoided when possible in general :)